recipes/pciutils/all/conanfile.py

```diff
@@ -1,76 +1,128 @@
+        if self.settings.compiler == "gcc":
+            return shutil.which(f"gcc-{compiler_version}") or shutil.which(f"gcc-{major}") or shutil.which("gcc") or ""
+        if self.settings.compiler == "clang":
+            return shutil.which(f"clang-{compiler_version}") or shutil.which(f"clang-{major}") or shutil.which("clang") or ""
+        return ""
 
-    def _make(self, targets):
+    def generate(self):
         yes_no = lambda v: "yes" if v else "no"
-        autotools = AutoToolsBuildEnvironment(self)
-        autotools.make(args=["SHARED={}".format(yes_no(self.options.shared)),
-                             "ZLIB={}".format(yes_no(self.options.with_zlib)),
-                             "HWDB={}".format(yes_no(self.options.with_udev)),
-                             "PREFIX={}".format(self.package_folder),
-                             "OPT={}".format("{} {}".format(
-                                 autotools.vars["CPPFLAGS"], autotools.vars["CFLAGS"])),
-                             "DNS=no"],
-                       target=" ".join(targets))
+        tc = AutotoolsToolchain(self)
+        env_vars = tc.environment().vars(self)
+        tc.make_args = [
+            f"SHARED={yes_no(self.options.shared)}",
+            f"ZLIB={yes_no(self.options.with_zlib)}",
+            f"HWDB={yes_no(self.options.with_udev)}",
+            f"DESTDIR={self.package_folder}",
+            "PREFIX=/",
+            "DNS=no",
+            f"CC={self._cc}",
+        ]
+        tc.generate()
+        deps = AutotoolsDeps(self)
+        deps.generate()
 
     def build(self):
-        with tools.chdir(self._source_subfolder):
-            self._make(["all"])
+        with chdir(self, self.source_folder):
+            autotools = Autotools(self)
+            autotools.make(target="all")
 
     def package(self):
-        with tools.chdir(self._source_subfolder):
-            self._make(["install", "install-pcilib"])
+        with chdir(self, self.source_folder):
+            autotools = Autotools(self)
+            autotools.make(target="install")
+            autotools.make(target="install-pcilib")
 
-        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
-        self.copy("*.h", src=self._source_subfolder, dst="include", keep_path=True)
+        copy(self, "COPYING",
+             src=self.source_folder,
+             dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "*.h",
+             src=self.source_folder,
+             dst=os.path.join(self.package_folder, "include"),
+             keep_path=True)
 
         if self.options.shared:
-            tools.rename(src=os.path.join(self._source_subfolder, "lib", "libpci.so.3.7.0"),
-                dst=os.path.join(self.package_folder, "lib", "libpci.so"))
+            # libpci.so.3 -> libpci.so
+            with chdir(self, os.path.join(self.package_folder, "lib")):
+                os.symlink("libpci.so.3", "libpci.so")
```

Comment:
Any specific reason for this to be a symlink?

Reply:
The project does not create one and test_package fails without it.